backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +63,128 @@ private void updatePersonFacility(Person person, UUID facilityId) {
+
+    if (facilityId == null) {
+      // READ_PATIENT_LIST does NOT seem right, probably an admin role to see all patients across
+      // facilities?
+      perms.add(UserPermission.READ_PATIENT_LIST);
+    }
+    if (isArchived) {
+      perms.add(UserPermission.READ_ARCHIVED_PATIENT_LIST);
+    }
+    if (searchTerm != null) {
+      perms.add(UserPermission.SEARCH_PATIENTS);
+    } else {
+      // because searchTerm is null, they are reading permissions across all patients
+      perms.add(UserPermission.READ_PATIENT_LIST);
+    }
+
+    // check all the permissions in one call.
+    return _auth.userHasPermissions(perms);
   }
 
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public long getArchivedPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), true);
+  // called by List function and Count function
+  public Specification<Person> buildFilterForListFunc(
+      UUID facilityId, boolean isArchived, String searchTerm) {
+    // build up filter based on params
+    Specification<Person> filter = isDeletedFilter(isArchived);
+    if (facilityId == null) { // admin call to get all users
+      filter = filter.and(inWholeOrganizationFilter());
+    } else {
+      filter = filter.and(inFacilityFilter(facilityId));
+    }
+
+    if (searchTerm != null && !searchTerm.isEmpty()) {
+      filter = filter.and(nameMatchesFilter(searchTerm));
+    }
+    return filter;
   }
 
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public List<Person> getAllArchivedPatients(int pageOffset, int pageSize) {
-    return _repo
-        .findAllByOrganization(
-            _os.getCurrentOrganization(), true, PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  /**
+   * @param facilityId If null, then it means across whole organization
+   * @param pageOffset Pagination offset is zero based
+   * @param pageSize How many results to return, zero will result in the default page size (large)
+   * @param isArchived Default is false. true will ONLY show deleted users
+   * @param searchTerm Null returns all users, any string will filter by first,middle,last names
+   *     that start with these characters. Case insenstive. If fewer than
+   * @return A list of matching patients.
+   */
+  public List<Person> getPatients(
+      UUID facilityId, int pageOffset, int pageSize, boolean isArchived, String searchTerm) {
+    if (pageOffset < 0) {
+      pageOffset = DEFAULT_PAGINATION_PAGEOFFSET;
+    }
+    if (pageSize < 1) {
+      pageSize = DEFAULT_PAGINATION_PAGESIZE;
+    }
+
+    if (searchTerm != null && searchTerm.trim().length() < MINIMUMCHARFORSEARCH) {
+      return List.of(); // empty list
+    }
+
+    // first check permissions
+    if (!checkPermissionsForListFunc(facilityId, isArchived, searchTerm)) {
```

Comment:
`checkPermissionsForListFunc()` should just become its own annotation `@RequireSpecificPatientSearchPermission` (or something) in `AuthorizationConfiguration`, with an associated function in `UserAuthorizationVerifier`. it can leverage variables like `RequireGlobalAdminUserOrPermissionManageTargetUser` does with `userId`. you can then apply that annotation to `getPatients` and `getPatientsCount` and remove the `UserAuthorizationVerifier` from this class

(also implied in this comment -- please change the name of this function to be a bit more descriptive, whether you convert it to an annotation or not. the annotation name above might be a good option?)

Reply:
updated the name and left a not as something to be changed in the future. trying to limit scope of this PR and get it into main before merging it becomes impossible (there are both server and FE react changes)